src/python/pants/init/logging.py

```diff
@@ -42,13 +42,70 @@ def _maybe_configure_extended_logging(logger):
     _configure_requests_debug_logging()
 
 
+def init_rust_logger(level):
+  native = Native()
+  levelno = get_numeric_level(level)
+  print("BL: {}, I want to init logging with level {}".format(os.getpid(), level))
+  native.init_rust_logging(levelno)
+
+
+def setup_logging_to_stderr(python_logger, level):
+  native = Native()
+  levelno = get_numeric_level(level)
+  handler = create_native_stderr_log_handler(levelno, native, stream=sys.stderr)
+  python_logger.addHandler(handler)
+  python_logger.setLevel("TRACE")
+
+
 def setup_logging_from_options(bootstrap_options):
   # N.B. quiet help says 'Squelches all console output apart from errors'.
   level = 'ERROR' if bootstrap_options.quiet else bootstrap_options.level.upper()
-  return setup_logging(level, console_stream=sys.stderr, log_dir=bootstrap_options.logdir)
+  native = Native()
+  return setup_logging(level, console_stream=sys.stderr, log_dir=bootstrap_options.logdir, native=native)
+
+
+class NativeHandler(StreamHandler):
+
+  def __init__(self, level, native=None, stream=None, native_filename=None):
+    super(NativeHandler, self).__init__(stream)
+    self.native = native
+    self.native_filename = native_filename
+    self.setLevel(level)
+
+  def emit(self, record):
+    msg = ("{}:{} - {}".format(record.name, os.getpid(), self.format(record)))
+    # msg = ("{}".format(self.format(record)))
+    self.native.write_log(msg, record.levelno, record.name)
+
+
+def create_native_pantsd_file_log_handler(level, native, native_filename):
+  try:
+    native.setup_pantsd_logger(native_filename, get_numeric_level(level))
+  except Exception as e:
+    print("Error setting up pantsd logger: {}".format(e), file=sys.stderr)
```

Comment:
Here, you certainly will want to use `sys.stderr`, because print is using unicode. This is right.

Reply:
Ack